xdsl/dialects/builtin.py

```diff
@@ -1031,23 +1031,28 @@ def __init__(self, types: list[Attribute] | ArrayAttr[Attribute]) -> None:
         super().__init__([types])
 
 
+_VectorTypeElement = TypeVar(
+    "_VectorTypeElement", bound=Attribute, covariant=True, default=Attribute
+)
+
+
 @irdl_attr_definition
 class VectorType(
-    Generic[AttributeCovT],
```

Comment:
Why would `AttributeCovT` not default to Attribute? There may be a downside I'm not appreciating, it just feels like it might make sense as an opt-out thing. My intuition is that since Attributes are immutable it's always safe to assume that the generic argument is covariant up to Attribute, and doesn't have to be Unknown

Reply:
We can do that if you prefer, I just didn't want to introduce a huge debate on that.

Reply:
I vote for this approach if you're also happy with it.

Reply:
Should be done now